### PR TITLE
Migration with add-on inconsistency

### PR DESCRIPTION
**Pull request by dzedro**
poo#18500

[migration_sle12_install_sdk_inconsistency](http://10.100.12.155/tests/6243#step/yast2_migration/27)
[migration_sle12_deactivate_sdk_inconsistency](http://10.100.12.155/tests/6244#step/yast2_migration/27)
[migration_sle12_install_we_inconsistency](http://10.100.12.155/tests/6245#step/yast2_migration/54)
[migration_sle12_deactivate_we_inconsistency](http://10.100.12.155/tests/6247#step/yast2_migration/52)

### CODE REVIEW

File: tests/online_migration/sle12_online_migration/yast2_migration.pm [hunk 1]
```diff
@@ -31,7 +31,51 @@ sub run {
         # set blank screen to be never for current session
         script_run("gsettings set org.gnome.desktop.session idle-delay 0");
         become_root;
+        if (check_var('HDDVERSION', '12')) {
```
Comment (okurz) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: does this specifically only apply for SLE 12 GA or for <= SLE 12 GA?

`if (!sle_version_at_least('12', version_variable => 'HDDVERSION')` should work in the latter case

File: tests/online_migration/sle12_online_migration/yast2_migration.pm [hunk 2]
```diff
@@ -31,7 +31,51 @@ sub run {
...
+            # use latest yast2-registration version because is not officially available
+            assert_script_run 'wget --quiet ' . data_url('yast2-registration-3.1.129.18-1.noarch.rpm');
+            zypper_call '--no-gpg-checks in yast2-registration-3.1.129.18-1.noarch.rpm';
+            record_info 'workaround', 'until the package is available for SLE 12 GA';
```
Comment (okurz) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: does this need to specified on the release notes or something? Doesn't the customer have the same problem?

File: tests/online_migration/sle12_online_migration/yast2_migration.pm [hunk 3]
```diff
@@ -51,23 +95,36 @@ sub run {
...
+                send_key 'alt-y';                       # yes
+            }
+            if (match_has_tag 'yast2-migration-inconsistency') {
+                wait_screen_change { send_key 'alt-c' };    # continue
+            }
+        }
+    }
+    assert_screen 'yast2-migration-target';
+    send_key "alt-p";                                       # focus on the item of possible migration targets
     send_key_until_needlematch 'migration-target-' . get_var("VERSION"), 'down', 5;
```
Comment (okurz) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: I guess you want `get_required_var('VERSION')` here

Comment (dzedro) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: funny you comment code I didn't touch

File: tests/online_migration/sle12_online_migration/yast2_migration.pm (hunk 1 above)
Comment (dzedro) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: it's only SLE 12 GA

File: tests/online_migration/sle12_online_migration/yast2_migration.pm (hunk 2 above)
Comment (dzedro) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: again, it's only SLE 12 GA, it is not available now for SLE 12 GA from official SCC or proxy SCC repo, so I have to install the changed yast2-registration to test the new behavior ...

File: tests/online_migration/sle12_online_migration/yast2_migration.pm (hunk 3 above)
Comment (okurz) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: Yeah, right. Ignore

**Comment by dzedro** [comment]
It should be done
PR for needles https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/385
Some tests http://10.100.12.155/tests
Only SP 1 migration is failing reboot just gets frozen, I will test it manually, maybe try new disk image.

**Comment by okurz** [comment]
+1

**Comment by okurz** [comment]
needles merged

**Comment by dzedro** [comment]
Replacement of SP1 disk image solved the failing SP1 test
http://19.100.12.155/tests/6419

File: tests/online_migration/sle12_online_migration/yast2_migration.pm (hunk 2 above)
Comment (mitiao) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: @dzedro This change causes the issue: https://progress.opensuse.org/issues/19522

Also I can see the version of yast2-registration is shown as following image if fully patch applied before migration on SLE12 GA:
screenshot_20170606_122259
Does it meet the requirement to test new behaviours? 
If so, `if (check_var('HDDVERSION', '12') && !get_var(FULL_UPDATE))` is better.

Comment (dzedro) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: Created PR to fix that, have to ask YaST why the update doesn't work when the package is there

**Pull request merged**